sql/src/main/java/io/crate/planner/operators/HashJoin.java

```diff
@@ -107,18 +103,34 @@ public ExecutionPlan build(PlannerContext plannerContext,
                                Row params,
                                Map<SelectSymbol, Object> subQueryValues) {
 
-        ExecutionPlan left = lhs.build(
+
+        ExecutionPlan leftExecutionPlan = lhs.build(
             plannerContext, projectionBuilder, NO_LIMIT, 0, null, null, params, subQueryValues);
-        ExecutionPlan right = rhs.build(
+        ExecutionPlan rightExecutionPlan = rhs.build(
             plannerContext, projectionBuilder, NO_LIMIT, 0, null, null, params, subQueryValues);
 
-        ResultDescription leftResultDesc = left.resultDescription();
-        ResultDescription rightResultDesc = right.resultDescription();
+        LogicalPlan leftLogicalPlan = lhs;
+        LogicalPlan rightLogicalPlan = rhs;
+
+        boolean tablesSwitched = false;
+        // Move smaller table to the right side
```

Comment:
why vs. what -  more meaningful comment would include that we do this because benchmarks indicate that it's faster.

Reply:
yep thanks, will change